test/com/google/enterprise/adaptor/database/JdbcFixture.java

```diff
@@ -16,28 +16,59 @@
 
 import static org.junit.Assert.assertTrue;
 
+import com.google.common.base.Strings;
 import org.h2.jdbcx.JdbcDataSource;
 
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.sql.Connection;
+import java.sql.DriverManager;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.ArrayDeque;
+import java.util.Properties;
 
 /** Manages an in-memory H2 database for test purposes. */
 class JdbcFixture {
-  public static final String DRIVER_CLASS = "org.h2.Driver";
-  public static final String URL =
-      "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DEFAULT_ESCAPE=";
-  public static final String USER = "sa";
-  public static final String PASSWORD = "";
+  public static String database;
+  public static String driver_class;
+  public static String url;
+  public static String user;
+  public static String password;
 
   private static ArrayDeque<AutoCloseable> openObjects = new ArrayDeque<>();
 
-  /**
-   * Gets a JDBC connection to a named in-memory database.
-   * <p>
-   * Connection options:
+  /*
+   * Initializes the database connection parameters from build.properties file.
+   * Defaults to H2 database.
+   */
+  public static void initialize() throws IOException {
+    String propertiesFile = System.getProperty("build.properties");
+    Properties properties = new Properties();
+    if (!Strings.isNullOrEmpty(propertiesFile)) {
+      try {
+        properties.load(new FileReader(propertiesFile));
+        if (Strings.isNullOrEmpty(properties.getProperty("test.database"))) {
+          loadDefaultProperies(properties);
```

Comment:
Just load the default properties first, or better, as I suggested, make them the default Properties object for the properties you are loading, or since you aren't using the properties directly, just avoid having default properties at all, and just set the fields to the default values if loading the properties failed.